tests/unit_tests/db_engine_specs/test_gsheets.py

```diff
@@ -0,0 +1,177 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=unused-argument, invalid-name
+from superset.errors import ErrorLevel, SupersetError, SupersetErrorType
+
+
+class ProgrammingError(Exception):
+    """
+    Dummy ProgrammingError so we don't need to import the optional gsheets.
+    """
+
+
+def test_validate_parameters_simple(mocker, app_context):
```

Comment:
I recommend to put all your tests under a Class so you can organize them more easily: skip, mark or annotate it with a fixture who will be applied for all the tests there

Reply:
It's not clearly documented but when we switched to `pytest` we decided to stop using classes and adopt a functional approach. There's some discussion here: https://github.com/apache/superset/pull/12680#discussion_r566120529